### PR TITLE
add ipynb - DQN

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
adding https://github.com/PyTorchLightning/pytorch-lightning/blob/master/notebooks/08-Domain-specific-demos.ipynb

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
facing https://github.com/computationalmodelling/nbval/issues/151#issuecomment-863022746
```
Notebook cell execution failed
Cell X: Timeout of 5 seconds exceeded waiting for output.
```

File: .actions/ipynb-test.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ if [ $accel == 1 ]
 then
   #python $py_file
   printf "Testing: $ipynb_file\n"
-  python -m pytest $ipynb_file -v --nbval
+  python -m pytest $ipynb_file -v --nbval --nbval-cell-timeout=300
```
Comment (Borda) [inline .actions/ipynb-test.sh:31]:
```suggestion
  python -m pytest $ipynb_file -v --nbval
```

File: lightning_examples/reinforce-learning-DQN/.meta.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+title: How to train a Deep Q Network
+author: PL team
+created: 2021-01-31
+updated: 2021-02-23
```
Comment (Borda) [inline lightning_examples/reinforce-learning-DQN/.meta.yml:4]:
```suggestion
updated: 2021-06-17
```

File: lightning_examples/reinforce-learning-DQN/.meta.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+license: CC
```
Comment (Borda) [inline lightning_examples/reinforce-learning-DQN/.meta.yml:5]:
```suggestion
license: CC BY-SA
```

**Review by tchaton — APPROVED** [review]
LGTM ! Should we remove tensorboad as it doesn't render ?

File: lightning_examples/reinforce-learning-DQN/dqn.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    Simple MLP network
+    """
+
+    def __init__(self, obs_size: int, n_actions: int, hidden_size: int = 128):
+        """
+        Args:
+            obs_size: observation/state size of the environment
+            n_actions: number of discrete actions available in the environment
+            hidden_size: size of hidden layers
+        """
```
Comment (ethanwharris) [inline lightning_examples/reinforce-learning-DQN/dqn.py:47]:
```suggestion
    """
    Simple MLP network
    
    Args:
        obs_size: observation/state size of the environment
        n_actions: number of discrete actions available in the environment
        hidden_size: size of hidden layers
    """

    def __init__(self, obs_size: int, n_actions: int, hidden_size: int = 128):
```

File: lightning_examples/reinforce-learning-DQN/dqn.py [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+            gamma: discount factor
+            sync_rate: how many frames do we update the target network
+            replay_size: capacity of the replay buffer
+            warm_start_size: how many samples do we use to fill our buffer at the start of training
+            eps_last_frame: what frame should epsilon stop decaying
+            eps_start: starting value of epsilon
+            eps_end: final value of epsilon
+            episode_length: max length of an episode
+            warm_start_steps: max episode reward in the environment
+        """
```
Comment (ethanwharris) [inline lightning_examples/reinforce-learning-DQN/dqn.py:238]:
```suggestion
    """Basic DQN Model

    Args:
        batch_size: size of the batches")
        lr: learning rate
        env: gym environment tag
        gamma: discount factor
        sync_rate: how many frames do we update the target network
        replay_size: capacity of the replay buffer
        warm_start_size: how many samples do we use to fill our buffer at the start of training
        eps_last_frame: what frame should epsilon stop decaying
        eps_start: starting value of epsilon
        eps_end: final value of epsilon
        episode_length: max length of an episode
        warm_start_steps: max episode reward in the environment
    """

    def __init__(
        self,
        batch_size: int = 16,
        lr: float = 1e-2,
        env: str = "CartPole-v0",
        gamma: float = 0.99,
        sync_rate: int = 10,
        replay_size: int = 1000,
        warm_start_size: int = 1000,
        eps_last_frame: int = 1000,
        eps_start: float = 1.0,
        eps_end: float = 0.01,
        episode_length: int = 200,
        warm_start_steps: int = 1000,
    ) -> None:
```

**Review by ethanwharris — APPROVED** [review]
LGTM, small comments

File: lightning_examples/reinforce-learning-DQN/dqn.py (hunk 4 above)
Comment (Borda) [reply lightning_examples/reinforce-learning-DQN/dqn.py:47]: is there a convention for this? I would tend to rather have the args explanation closer to the method itself...

Comment (ethanwharris) [reply lightning_examples/reinforce-learning-DQN/dqn.py:47]: That would be fine, although then should be consistent and do that for all classes in the notebook haha 😆

**Pull request merged**